File: Week3/Code/get_TreeHeight.R
#!/usr/bin/env Rscript
# Amy Solman amy.solman19@imperial.ac.uk
# 2nd December 2019
# get_TreeHeigh.R: This function takes a csv file name from the command
#line, calculates heights of trees given distance to each tree
# base and angle to its top, using the trigonometric formula
# with input filename included in output filename


args <- commandArgs(TRUE)
  
MyTrees <- read.table(args[1], sep=",", stringsAsFactors = FALSE, header=TRUE) # read trees.csv to MyTrees data frame

TreeDistance <- as.numeric(MyTrees[[2]]) # assign second column of MyTrees data frame to TreeDistance numerical vector

TreeDegrees <- as.numeric(MyTrees[[3]]) # assign third column of MyTrees data frame to TreeDegrees numerical vector

TreeHeight <- function(degrees, distance){ # defines the function and the two arguments it will take
  radians <- degrees * pi / 180 # applies 'degrees' multiplied by pie divided by 180 to radians
  height <- distance * tan(radians) 
  print(paste(height))
}

Tree.Height <- TreeHeight(TreeDegrees, TreeDistance)

# Add height to MyTrees data frame
MyTrees$"Tree.Height.m" <- Tree.Height

filename <- paste0(tools::file_path_sans_ext(args[1]),"_treeheights.csv")
output_folder <- paste0("../Results/",filename) 

write.table(MyTrees, output_folder, row.names=FALSE, col.names=TRUE)

File: Week3/Code/get_TreeHeight.py
#!/usr/bin/env python3
# Date: Dec 2019

"""Takes input file of tree distances and degress, calculates height
and outputs results file with input file name."""

__appname__ = 'get_TreeHeight.py'
__author__ = 'Amy Solman (amy.solman19@imperial.ac.uk)'
__version__ = '0.0.1'

import sys
import pandas as pd
import os
import math


MyDF = pd.read_csv(sys.argv[1], sep=',')

inputfile = sys.argv[1]

TreeDistance = MyDF.iloc[:, 1]


TreeDegrees = MyDF.iloc[:, 2]

def Tree_Height(degrees, distance):
    radians = degrees*math.pi/180
    height = distance*radians.astype(float).map(math.tan)
    return(height)

TreeHeight = Tree_Height(TreeDegrees, TreeDistance)

MyDF['TreeHeight'] = TreeHeight

file_dir = "../Results"
outputfile = inputfile.split(".")[0] + "_treeheights_py.csv"
final_path = os.path.join(file_dir, outputfile)

export_csv = MyDF.to_csv(final_path, header=True, index=False)
